### ISSUE TITLE
local variable 'values' referenced before assignment

### ISSUE THREAD

**Issue by Sanket-DataEnt** [pytorch/pytorch#36313]
While performing OneCycleLR, i was tryinh to first use LR function lambda and got the followimng error : 

# Experiment parameters
lr_find_epochs = 5
start_lr = 1e-7
end_lr = 0.1

# Set up the model, optimizer and loss function for the experiment
model = model.to(device)

optimizer = torch.optim.SGD(model.parameters(), lr = start_lr)
criterion = nn.CrossEntropyLoss()

# LR function lambda

lr_lambda = lambda x: math.exp(x * math.log(end_lr / start_lr) / (lr_find_epochs * len(dataloaders["train"])))
scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lr_lambda)

ERROR : 
UnboundLocalError                         Traceback (most recent call last)

<ipython-input-22-faae7279e1d9> in <module>()
      1 
      2 lr_lambda = lambda x: math.exp(x * math.log(end_lr / start_lr) / (lr_find_epochs * len( dataloaders["train"])))
----> 3 scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lr_lambda)

2 frames

/usr/local/lib/python3.6/dist-packages/torch/optim/lr_scheduler.py in step(self, epoch)
    148                     values = self.get_lr()
    149 
--> 150         for param_group, lr in zip(self.optimizer.param_groups, values):
    151             param_group['lr'] = lr
    152 

UnboundLocalError: local variable 'values' referenced before assignment

cc @vincentqb

**Comment by vincentqb** [comment]
What version do you have? Have you tried with the latest code on master? (in particular, after #32997)

**Comment by zml24** [comment]
I meet the same bug here.

`scheduler = LambdaLR(optimizer, lr_lambda=lambda step: (args.d_model ** -0.5) * min(step ** -0.5, step * args.warmup_step ** -1.5))`

And there are 
```
Traceback (most recent call last):
  File "main.py", line 193, in <module>
    main()
  File "main.py", line 183, in main
    train(args, model, optimizer, train_dataloader, train_type='s2t', align=False)
  File "/data/data0/zml/git/bridge/utils.py", line 169, in train
    scheduler = LambdaLR(optimizer, lr_lambda=lambda step: (args.d_model ** -0.5) * min(step ** -0.5, step * args.warmup_step ** -1.5))
  File "/data/data0/zml/anaconda3/envs/torch1.4/lib/python3.6/site-packages/torch/optim/lr_scheduler.py", line 189, in __init__
    super(LambdaLR, self).__init__(optimizer, last_epoch)
  File "/data/data0/zml/anaconda3/envs/torch1.4/lib/python3.6/site-packages/torch/optim/lr_scheduler.py", line 77, in __init__
    self.step()
  File "/data/data0/zml/anaconda3/envs/torch1.4/lib/python3.6/site-packages/torch/optim/lr_scheduler.py", line 150, in step
    for param_group, lr in zip(self.optimizer.param_groups, values):
UnboundLocalError: local variable 'values' referenced before assignment
```

torch==1.4.0
cuda version==10.1

**Comment by vincentqb** [comment]
Please update to the latest version (1.5.0). This issue has been fixed by #32997. I will close this issue, but please feel free to re-open if the issue continues in the latest version.
